cerberus/forms.py

```diff
@@ -77,18 +54,128 @@ class Meta:
 
 
 class BookingForm(forms.ModelForm):
+    attributes = {
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **issue (complexity):** Consider separating frontend logic from Django forms to maintain a clear separation of concerns.

The introduction of frontend logic directly into Django forms, particularly through attributes like `x-data`, `x-model.number.fill`, `@change`, etc., significantly increases the complexity of the codebase. This approach mixes backend form logic with frontend behavior, leading to a blurring of the separation of concerns. It makes the code harder to maintain, as developers now need to navigate both Python and JavaScript within the same files. Additionally, the use of custom widgets and attributes for frontend interactions further complicates the form definitions.

A more maintainable approach would be to keep the frontend logic separate from the Django forms. Instead of embedding JavaScript directly into the form widgets, consider using data attributes minimally to mark elements for frontend interaction and handling the behavior entirely in separate JavaScript files. This maintains a clear separation of concerns, simplifies the form definitions, and reduces the cognitive load on developers by keeping backend and frontend logic distinct.